### PR TITLE
Support new annotations (and change of instrumentation name for opentelemetry-annotations)

### PR DESCRIPTION
**Pull request by trask**
Part of #6245

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-annotations-1.16/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/otelannotations/WithSpanInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            InstrumentationConfig.get().getString(TRACE_ANNOTATED_METHODS_EXCLUDE_CONFIG));
+    for (Map.Entry<String, Set<String>> entry : excludedMethods.entrySet()) {
+      String className = entry.getKey();
+      ElementMatcher.Junction<ByteCodeElement> classMather =
+          isDeclaredBy(ElementMatchers.named(className));
+
+      ElementMatcher.Junction<MethodDescription> excludedMethodsMatcher = none();
+      for (String methodName : entry.getValue()) {
+        excludedMethodsMatcher = excludedMethodsMatcher.or(ElementMatchers.named(methodName));
+      }
```
Comment (mateuszrzeszutek) [inline instrumentation/opentelemetry-instrumentation-annotations-1.16/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/otelannotations/WithSpanInstrumentation.java:106]: Nit: could be using `ElementMatchers.namedOneOf()`

File: instrumentation/opentelemetry-instrumentation-annotations-1.16/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/otelannotations/WithSpanInstrumentationModule.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import java.util.List;
+
+/** Instrumentation for methods annotated with {@link WithSpan} annotation. */
+@AutoService(InstrumentationModule.class)
+public class WithSpanInstrumentationModule extends InstrumentationModule {
+
+  public WithSpanInstrumentationModule() {
+    super("opentelemetry-annotations");
```
Comment (mateuszrzeszutek) [inline instrumentation/opentelemetry-instrumentation-annotations-1.16/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/otelannotations/WithSpanInstrumentationModule.java:21]: Just to make sure: this is the same name as the extension annotations, is that correct? (I think it makes sense for these two modules to have the same name, just wanted to confirm that this is on purpose, and not copy-pasta)

File: instrumentation/opentelemetry-instrumentation-annotations-1.16/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/otelannotations/WithSpanParameterAttributeNamesExtractor.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    try {
+      return AnnotationReflectionHelper.bindAnnotationElementMethod(
+          MethodHandles.lookup(), spanAttributeAnnotation, "value", String.class);
+    } catch (Throwable exception) {
+      return annotation -> "";
+    }
+  }
+
+  @Override
+  @Nullable
```
Comment (mateuszrzeszutek) [inline instrumentation/opentelemetry-instrumentation-annotations-1.16/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/otelannotations/WithSpanParameterAttributeNamesExtractor.java:46]: Always returns non-null value
```suggestion
```

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-annotations-1.16/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/otelannotations/WithSpanInstrumentationModule.java (hunk 2 above)
Comment (trask) [reply instrumentation/opentelemetry-instrumentation-annotations-1.16/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/otelannotations/WithSpanInstrumentationModule.java:21]: It was definitely copy-pasta, I will rename to match the module names

**Pull request merged**